ml-agents/mlagents/trainers/demo_loader.py

```diff
@@ -93,17 +93,10 @@ def demo_to_buffer(
     demo_buffer = make_demo_buffer(info_action_pair, behavior_spec, sequence_length)
     if expected_behavior_spec:
         # check action dimensions in demonstration match
-        if behavior_spec.action_shape != expected_behavior_spec.action_shape:
+        if behavior_spec.action_spec != expected_behavior_spec.action_spec:
```

Comment:
Maybe not in this PR, but maybe we can have __eq__ defined on behavior_spec as well so this whole code is simplified?

Reply:
I originally implemented it this way but then I lost some granularity in the error messages I could raise i.e. whether or not the offending attribute was action_spec or obs shapes.